util/fs/fs.go

```diff
@@ -43,11 +54,54 @@ func Detect(devicePath string) (string, error) {
 // resize2fs only runs a resize when it is
 // required on the device; otherwise, it just exits with a code 0 and a message.
 func Resize(devicePath string) error {
-	cmd := exec.Command("sudo", "-n", "resize2fs", "-f", devicePath)
+	fsType, err := Detect(devicePath)
+	if err != nil {
+		return err
+	}
+
+	switch fsType {
+	case "ext2", "ext3", "ext4":
+	default:
+		return ErrResizeNotAvailable
+	}
+
+	cmd, err := sudoCmd("resize2fs", "-f", devicePath)
+	if err != nil {
+		return err
+	}
 	output, err := cmd.CombinedOutput()
 	output = bytes.Trim(output, "\r\n \t")
 	if err != nil {
-		return fmt.Errorf("Resize: %s: %s", devicePath, string(output))
+		return fmt.Errorf("Resize: %v: %v", devicePath, string(output))
 	}
 	return nil
 }
+
+func sudoCmd(name string, args ...string) (*exec.Cmd, error) {
+	prefix, err := sudoCmdPrefix()
+	if err != nil {
+		return nil, err
+	}
+	if len(prefix) == 1 {
+		args = append([]string{name}, args...)
+		name = prefix[0]
+	} else if len(prefix) > 1 {
+		args = append(append(prefix[1:], name), args...)
+		name = prefix[0]
+	}
+	cmd := exec.Command(name, args...)
+	return cmd, nil
+}
+
+// sudoCmdPrefix accounts for sudo only being necessary when UID != 0 (i.e. when not
+// root).
+func sudoCmdPrefix() ([]string, error) {
```

Comment:
Not sure I understand the proposal here.

The signature `exec.Command(name string, args ...string)` requires a binary available in from `$PATH`, so a string of `sudo -n foo` would trigger an error since no such executable exists.